controllers/ovndbcluster_controller.go

```diff
@@ -566,21 +592,77 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 	)
 	if err == nil && len(svcList.Items) > int(*(instance.Spec.Replicas)) {
 		for i := len(svcList.Items) - 1; i >= int(*(instance.Spec.Replicas)); i-- {
+			fullServiceName := fmt.Sprintf("%s-%d", serviceName, i)
 			svcLabels := map[string]string{
 				common.AppSelector:                   serviceName,
-				"statefulset.kubernetes.io/pod-name": serviceName + fmt.Sprintf("-%d", i),
+				"statefulset.kubernetes.io/pod-name": fullServiceName,
 			}
 			err = service.DeleteServicesWithLabel(
 				ctx,
 				helper,
 				instance,
 				svcLabels,
 			)
+			if err != nil {
+				err = fmt.Errorf("Error while deleting service with name %s: %w", serviceName, err)
+				return ctrl.Result{}, err
+			}
+			// Delete DNS records for deleted services/pods
+			namespace := helper.GetBeforeObject().GetNamespace()
+			err = deleteDNSData(ctx, helper, fullServiceName, namespace)
```

Comment:
the dnsdata no longer exists for fulServiceName so this call and definition of function deleteDNSData not needed.

Reply:
No, it does exist. 
For each service a DNSData will be created. Inside this DNSData entry it will be the resolution `ovsdbserver-sb.openstack.svc -> SB_POD_INTERNALAPI_IP`
But at the end you'll have the same number of entries as services. What it was deleted was the resolution `ovsdbserver-sb-X.openstack.svc -> SB_POD_INTERNALAPI_IP`

Reply:
Okk i expected it to be single dnsdata per service from past discussion, keeping dnsdata per service just looks unnecessary at this point.

Reply:
It's something that can be done, but it would miss dp3. Can create jira to do a follow up if necessary